### ISSUE TITLE
Issues mapping an existing entry/user

### ISSUE THREAD

**Issue by dbrown199747** [verbb/formie#1732]
### Describe the bug

I want to update an existing entry with some form data, however I can't get it to update an existing entry, I can only get it to create new entries with the field mapping.

I've attemted to set this up, but can't get it to work. Once i've set it up and submitted, the form log tells me that I have to map the title in the 'Entry Attribute Mapping' section as this is required. I have also enabled the 'Update Entries' light-box and mapped the relevant fields, and tried enabling the 'Overwrite Content' lightbox but I don't see what these do, as every time it creates a new entry instead of updating an existing entry.

Are you able to tell me what i'm doing wrong? I've also tried this with Users, but I get a logged error saying that the username or email already exists, but I have to enter a username and email because they are required fields.

See screenshots of how i've set the mapping up.

<img width="1620" alt="screenshot 1" src="https://github.com/verbb/formie/assets/83223160/955d2904-2493-4645-b5ac-cce1c9a3f627">

<img width="1627" alt="screenshot 2" src="https://github.com/verbb/formie/assets/83223160/c518dd11-e264-49e5-b852-e8f2c2f8224f">

<img width="1618" alt="screenshot 3" src="https://github.com/verbb/formie/assets/83223160/adc61155-735d-4806-8abf-c15e1ecbe517">

### Steps to reproduce

1. 
2. 

### Form settings

- Multi-page form: Yes or No
- Submission Method: Ajax or Page Reload
- Client-side Validation: Yes or No
- Custom Form Templates: Yes or No

### Craft CMS version

Craft Pro 4.7.3

### Plugin version

2.1.3

### Multi-site?

No

**Comment by engram-design** [comment]
So that all looks correct to me. The values you map under "Update Element Mapping" mean that it will take the values you map, compare them to any existing entry, and if so, update those. So you're mapping "Current Users Username" in Formie to the "Testing" field in your entry.

Can I ask what the "Current Users Username" Formie field type is? What sort of data is that? If you look at the resulting entry, is that populated correctly? And what sort of field is "Testing" in your entry?

**Comment by dbrown199747** [comment]
Thanks for getting back to me.

I basically added a 'testing' field just to try and output a value into it. The 'Current Users Username' is a hidden field that i've set the value to be the logged in users username.

What i'm trying to do is:
1. Have a quiz form on a page
2. In the CMS for the page that the quiz is on, there will be a 'users completed quiz' field
3. When submitting the form, I want to map the current user into the 'users completed quiz' field, so that I can have a message on the page saying 'you have already completed this quiz...'

Are you able to give me an idea of how to do this? Currently I can only get Formie to create new entries when using the 'Elements' integration.

Thanks,
Dan

**Comment by engram-design** [comment]
So comparing the value of fields in your entry and submission should be working as you've setup. I've also just confirmed that's working for me on a similar setup.

Under the hood, it's going to create a query like `Entry::find()->myCustomField('myValue')->one()` to find an existing entry, where `myCustomField` is the handle of the entry field, and `myValue` is the content for the field you've set to map to `myCustomField`.

I'm wondering if you could simplify your use-case where if all you want to do is just show a message if the user has filled out the form already, you could query for a submission for that form for that user.

```twig
{% submission = craft.formie.submissions().form('myQuiz').email(currentUser.email).one() %}

{% if submission %}
    You've already filled out this form
```

This would require a `email` Formie field added to your form, which could be a hidden input to record the email or username or user ID (whatever you want to identify a unique user) to check for an existing submission from.

**Comment by siebird** [comment]
I'm trying to do something similar – update two fields in an entry. I'm using the entries field in Formie to list available entries to update. 

I want to use the selected entry to map and update that entry. Within the update element mapping section, I'm only seeing custom fields to check for in Formie `3.0.0-beta 13`. Missing title, entry id, slug, etc.

**Comment by engram-design** [comment]
@siebird That would be a bug, unrelated to this. Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.0-beta.13"`.

**Comment by engram-design** [comment]
Fixed in [3.0.0-beta.14](https://github.com/verbb/formie/releases/tag/3.0.0-beta.14)

**Comment by siebird** [comment]
I always appreciate your responsiveness, Josh. Thanks!

**Comment by siebird** [comment]
I'm getting an error on the integration:

`Failed to trigger integration: ["Element integration failed for submission “7436”. Error: Cannot assign array to property craft\\base\\Element::$title of type ?string \/vendor\/verbb\/formie\/src\/integrations\/elements\/Entry.php:219"].`

It seems the entry field is passing an array. I would assume we could select an attribute to map to similar to the name field (ie. entry: id, entry: title, etc.)

**Comment by engram-design** [comment]
What are you mapping to `title`? What sort of Formie field is it?

**Comment by siebird** [comment]
I created a formie form to allow non-admin users to update entries in a channel. It has three fields:

- entries (a select menu to pick the entry they want to update)
- two text fields (to update two fields in the entry selected above)

I need to map the updated information back to the entry selected in the entries field.

**Comment by engram-design** [comment]
Can you send through screenshots of your mapping? Maybe you're mapping an Entries Formie field to the Title attribute of an entry? It might be better through a new issue, or via `support@verbb.io` as well (or the Craft Discord).

**Comment by engram-design** [comment]
Fixed in [2.1.20](https://github.com/verbb/formie/releases/tag/2.1.20) and [3.0.0-beta.16](https://github.com/verbb/formie/releases/tag/3.0.0-beta.16)
